UM/Settings/DefinitionContainer.py

```diff
@@ -332,16 +332,52 @@ def deserialize(self, serialized: str, file_name: Optional[str] = None) -> str:
         self._metadata["version"] = self.Version #Guaranteed to be equal to what's in the parsed data by the validation.
         self._metadata["container_type"] = DefinitionContainer
 
-        for key, value in parsed["settings"].items():
-            definition = SettingDefinition(key, self, None, self._i18n_catalog)
+        self._deserializeDefinitions(parsed["settings"])
+        return serialized
+
+    def _deserializeDefinitions(self, settings_dict: Dict[str, Any], force_category: Optional[str] = None) -> None:
+
+        # When there is a forced category (= parent) present, find the category parent, create it if it doesn't exist.
+        category_parent = None
+        if force_category:
+            category_parent = self.findDefinitions(key = force_category)
+            category_parent = category_parent[0] if len(category_parent) > 0 else None
+
+        for key, value in settings_dict.items():
+            definition = SettingDefinition(key, self, category_parent, self._i18n_catalog)
             self._definition_cache[key] = definition
             definition.deserialize(value)
-            self._definitions.append(definition)
+            if category_parent:
+                # Forced category; these are then children of that category, instead of full categories on their own.
+                category_parent.children.append(definition)
+            else:
+                self._definitions.append(definition)
 
         for definition in self._definitions:
             self._updateRelations(definition)
 
-        return serialized
+    def appendAdditionalSettingDefinitions(self, additional_settings: Dict[str, Dict[str, Any]]) -> None:
```

Comment:
Needs a bit of documentation. What is this doing? In what cases should one use this?